controllers/player.controller.js

```diff
@@ -1,43 +1,90 @@
 const Player = require('../models/player.model')
 
+const logError = requestId => {
+  console.log(
+    JSON.stringify({
+      kinto_request_id: requestId,
+      error: error
+    })
+  )
+}
+
 exports.hello = (req, res) => {
-  res.send('Hello world')
+  const name = req.params.name
+  res.send(`Hello ${name}`)
 }
 
-exports.player_create = (req, res) => {
-  let player = new Player({
-    name: req.body.name,
-    score: req.body.score
-  })
+exports.player_create = async (req, res) => {
+  try {
+    let player
+    player = await Player.find({ name: req.body.name })
 
-  player.save(function(err) {
-    if (err) {
-      return next(err)
+    if (!player.length) {
+      player = new Player({
+        name: req.body.name,
+        score: req.body.score
+      })
     }
-    res.send('Player Created successfully')
-  })
+    await player.save()
+    res.status(200).send(player)
+  } catch (error) {
+    const requestId = req.get('kinto-request-id')
+    logError(requestId)
+    res.send({ error: `player not saved: ${error}` })
+  }
+}
+
+exports.player_details = async (req, res) => {
+  try {
+    let player = await Player.find({ name: req.params.username })
+    const foundPlayer = player[0]
+    player = await Player.findById(foundPlayer._id)
```

Comment:
why are u doing to finds here? the `findById` is not needed, and instead of using `.find` use `.findOne` if you know there will be only one item